hal/inc/core_hal.h

```diff
@@ -284,6 +285,100 @@ typedef void(*HAL_Event_Callback)(int event, int flags, void* data);
 
 void HAL_Set_Event_Callback(HAL_Event_Callback callback, void* reserved);
 
+
+#define HAL_CORE_SLEEP_VERSION 2
+
+typedef enum hal_core_wakeup_source_type_t {
```

Comment:
Do you think it makes sense to move all these definitions into a separate header (as well as implementation into a separate source file) to keep `core_hal` clean?

Reply:
It makes sense I think. I was going to introduce a class to cache and restore peripherals' configurations, that's not allowed in c file though.

Reply:
> to cache and restore peripherals' configurations

See https://github.com/particle-iot/device-os/pull/1965#discussion_r345664545 about that. I'm not sure that should be part of our initial Sleep 2.0 offering.